nixos/modules/misc/nixpkgs.nix

```diff
@@ -42,12 +44,47 @@ let
     merge = lib.mergeOneOption;
   };
 
-  _pkgs = import ../../.. config.nixpkgs;
+  pkgsType = mkOptionType {
+    name = "nixpkgs";
+    description = "An evaluation of Nixpkgs; the top level attribute set of packages";
+    check = builtins.isAttrs;
+  };
 
 in
 
 {
   options.nixpkgs = {
+
+    pkgs = mkOption {
+      defaultText = ''
+        Situation-dependent; plain NixOS sets the option default to
+        <code>import nixpkgs { inherit config overlays system; }</code>,
+        where <code>nixpkgs</code> is the copy of Nixpkgs that corresponds
+        to this NixOS distribution; it is part of the same directory tree.
+        The arguments are taken from the configuration options
+        <code>config.nixpkgs.config</code>,
+        <code>config.nixpkgs.overlays</code>, etc.
+        ''
+        ;
+      default = import ../../.. { inherit (cfg) config overlays system; };
```

Comment:
How does it render in the documentation? Isn't the `defaultText` supposed to provide a literal example of the `default` value?

Reply:
That didn't look so good. Can't just move documentation around.

Now I've removed the docbook tags and added `literalExample` to remove the quotes. It's a code literal now, which is almost good. I can't put code like `import ../../../ { whatever}` without an explanation there. That'd be too vague.
So not perfect, but at least the formatting is a lot less wrong now.

Reply:
Maybe add the rest as part of the documentation attribute, and/or use `<nixos>` instead of `../../..` even if this is not exactly true.

Reply:
I've introduced a fake variable nixos to indicate the relative path. Not bad after all. Thanks for the input.